### PR TITLE
Create method that returns selected DbItem

### PR DESCRIPTION
**Pull request by norascheuch**
When running a query we need to know which DbItem is selected in the new DB panel. This PR adds a method to the DbManager that returns the currently selected item.

## Checklist

- [ ] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [ ] Issues have been created for any UI or other user-facing changes made by this pull request.
- [ ] _[Maintainers only]_ If this pull request makes user-facing changes that require documentation changes, open a corresponding docs pull request in the [github/codeql](https://github.com/github/codeql/tree/main/docs/codeql/codeql-for-visual-studio-code) repo and add the `ready-for-doc-review` label there.

### CODE REVIEW

File: extensions/ql-vscode/src/databases/db-manager.ts [hunk 1]
```diff
@@ -18,6 +19,16 @@ export class DbManager {
     });
   }
 
+  public selectedDbItem(): DbItem | undefined {
```
Comment (charisk) [inline extensions/ql-vscode/src/databases/db-manager.ts:22]: Can we name this `getSelectedDbItem` instead? 

I think a general rule we follow is to have verbs in function names. I don't think this is necessarily a hard and fast rule through the codebase, but it's definitely something we follow in this class.

**Review by charisk — COMMENTED** [review]
This is great, thank you! Love seeing all the tests too.

File: extensions/ql-vscode/src/databases/db-manager.ts (hunk 1 above)
Comment (norascheuch) [reply extensions/ql-vscode/src/databases/db-manager.ts:22]: Great point! I wondered if it's bad to have two methods with the exact same name but agree that `getSelectedDbItem` makes more sense!

File: extensions/ql-vscode/test/pure-tests/databases/db-item-selection.test.ts [hunk 2]
```diff
@@ -0,0 +1,299 @@
+import { DbItem, DbItemKind } from "../../../src/databases/db-item";
+import { getSelectedDbItem } from "../../../src/databases/db-item-selection";
+describe("db item selection", () => {
+  it("return undefined if no item is selected", () => {
+    const dbItems: DbItem[] = [
+      {
+        kind: DbItemKind.RootRemote,
+        children: [
+          {
```
[the remark this answers is not in the document]
Comment (norascheuch) [reply extensions/ql-vscode/test/pure-tests/databases/db-item-selection.test.ts:9]: it's a very good idea..

File: extensions/ql-vscode/test/pure-tests/databases/db-item-selection.test.ts [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+        },
+        {
+          kind: DbItemKind.RemoteRepo,
+          repoFullName: "owner1/repo3",
+          selected: false,
+        },
+      ],
+      selected: true,
+    });
+  });
```
[the remark this answers is not in the document]
Comment (norascheuch) [reply extensions/ql-vscode/test/pure-tests/databases/db-item-selection.test.ts:298]: 👍

File: extensions/ql-vscode/test/pure-tests/databases/db-item-selection.test.ts [hunk 4]
```diff
@@ -0,0 +1,299 @@
+import { DbItem, DbItemKind } from "../../../src/databases/db-item";
+import { getSelectedDbItem } from "../../../src/databases/db-item-selection";
+describe("db item selection", () => {
+  it("return undefined if no item is selected", () => {
```
[the remark this answers is not in the document]
Comment (norascheuch) [reply extensions/ql-vscode/test/pure-tests/databases/db-item-selection.test.ts:4]: 👍

File: extensions/ql-vscode/src/databases/db-item-selection.ts [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+export function getSelectedDbItem(dbItems: DbItem[]): DbItem | undefined {
+  for (const dbItem of dbItems) {
+    if (
+      dbItem.kind === DbItemKind.RootRemote ||
+      dbItem.kind === DbItemKind.RootLocal
+    ) {
+      for (const child of dbItem.children) {
+        switch (child.kind) {
+          case DbItemKind.LocalList:
+            if (child.selected) {
```
[the remark this answers is not in the document]
Comment (norascheuch) [reply extensions/ql-vscode/src/databases/db-item-selection.ts:12]: 🤦‍♀️

File: extensions/ql-vscode/test/pure-tests/databases/db-item-selection.test.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (norascheuch) [reply extensions/ql-vscode/test/pure-tests/databases/db-item-selection.test.ts:298]: I just realized we already have this case in line 88, I renamed the test to make it more visible

[the remark this answers is not in the document]
Comment (charisk) [reply extensions/ql-vscode/test/pure-tests/databases/db-item-selection.test.ts:193]: Oh oops no I think that's enough, sorry I missed it!

**Review by charisk — APPROVED** [review]
Thanks for making the changes and thinking about readability.

**Pull request merged**